test/solidity/simple-auction/simple-auction.js

```diff
@@ -0,0 +1,118 @@
+const { expect } = require('chai')
+const { ethers } = require('hardhat')
+const Constants = require('../../constants')
+const Utils = require('../../hts-precompile/utils')
+
+const sleep = (ms) => new Promise((r) => setTimeout(r, ms));
+
+describe('Solidity Simple Auction example', function () {
+  let factory, signers, wallet, wallet1, 
+      contract, hasError, bidAmount,
+      contractShortLived, bidAmountSmall, initialEvent
+  const CONTRACT_DURATION = 10000000000
+  const CONTRACT_SHORT_DURATION = 1
+  const TRANSACTION_VALUE = '100'
+  const TRANSACTION_VALUE_SMALL = '10'
+
+  before(async function () {
+    signers = await ethers.getSigners()
+    wallet = signers[0]
+    wallet1 = signers[1]
+
+    factory = await ethers.getContractFactory(Constants.Contract.SimpleAuction)
+    contractShortLived = await factory.deploy(CONTRACT_SHORT_DURATION, wallet1.address)
+    await contractShortLived.deployed();
+
+    bidAmount = ethers.utils.parseUnits(TRANSACTION_VALUE, 'gwei')
+    bidAmountSmall = ethers.utils.parseUnits(TRANSACTION_VALUE_SMALL, 'gwei')
+    await sleep(5000)
+  })
+
+  beforeEach(async function () {
+    hasError = false
+    contract = await factory.deploy(CONTRACT_DURATION, wallet.address)
+    await contract.deployed();
+
+    const trx = await contract.bid({ value: bidAmountSmall })
+    const receipt = await trx.wait()
+    initialEvent = receipt.events[0].event
+  })
+
+  it('should confirm "bid" function works', async function () {    
+    const highestBid = await contract.highestBid()
+    const highestBidder = await contract.highestBidder()
+
+    expect(highestBid.mul(Utils.tinybarToWeibarCoef)).to.equal(bidAmountSmall)
+    expect(highestBidder).to.equal(wallet.address)
+    expect(initialEvent).to.equal('HighestBidIncreased')
+  })
+
+  it('should confirm bid not high enough scenario works: BidNotHighEnough', async function () {
+    try {
+      const tr = await contract.bid({ value: bidAmountSmall })
+      const receipt = await tr.wait();
+      expect(receipt).to.not.be.null
+    } catch (error) {
+      hasError = true
```

Comment:
I am not sure this is a best practice/good way of writing test - using try/catch and hasError boolean value. I know we do it in some places, but does not seem the best way to do it. Personally, I use chaiAsPromised, you can see and example in my PR for the Blind Auction example - https://github.com/hashgraph/hedera-smart-contracts/blob/02b716d6a4a81d6cc83980c095931c420f311ce2/test/solidity/blind-auction/blindAuction.js#L203

@georgi-l95 @Nana-EC @natanasow @Ivo-Yankov what do you guys think?